nornir_nautobot/plugins/tasks/dispatcher/default.py

```diff
@@ -34,21 +34,57 @@ class DispatcherMixin:
     """Mixin for non-network driver related tasks."""
 
     tcp_port = 22
+    config_injections = []
+    config_end_string = ""
 
     @classmethod
     def _get_hostname(cls, task: Task, obj=None) -> str:  # pylint: disable=unused-argument
         return task.host.hostname
 
     @classmethod
-    def _get_tcp_port(cls, obj) -> str:
-        custom_field = obj.cf.get("tcp_port")
-        if isinstance(custom_field, int):
-            return custom_field
-        config_context = obj.get_config_context().get("tcp_port")
-        if isinstance(config_context, int):
-            return config_context
+    def _get_tcp_port(cls, obj) -> int:
+        # NautobotORMInventory (nautobot-plugin-nornir) inventory.
+        if hasattr(obj, "cf"):
+            custom_field = obj.cf.get("tcp_port")
+            if isinstance(custom_field, int):
+                return custom_field
+        if hasattr(obj, "get_config_context"):
+            config_context = obj.get_config_context().get("tcp_port")
+            if isinstance(config_context, int):
+                return config_context
+        # NautobotInventory (nornir-nautobot) inventory.
+        if obj.data["pynautobot_dictionary"].get("custom_fields"):
```

Comment:
Should this be a `obj.data.get("pynautobot_dictionary", {})`?

Reply:
Actually, may need to be:
```python
        if hasattr(obj, "data") and obj.data["pynautobot_dictionary"].get("custom_fields"):
```

Reply:
my perspective is pynautobot_dictionary (and data for that matter) is guaranteed to be there since the inventory plugin is creating that key.  Happy to still add that, but just providing my perspective.